sim-lib/src/lib.rs

```diff
@@ -918,8 +936,23 @@ async fn produce_events<N: DestinationGenerator + ?Sized, A: PaymentGenerator +
     sender: Sender<SimulationEvent>,
     listener: Listener,
 ) -> Result<(), SimulationError> {
+    let mut current_count = 0;
     loop {
-        let wait = node_generator.next_payment_wait();
+        if let Some(c) = node_generator.payment_count() {
+            if c == current_count {
+                log::info!(
+                    "Payment count has been met for {source}: {c} payments. Stopping the activity."
+                );
+                return Ok(());
+            }
+        }
+
+        let wait: Duration = if current_count == 0 {
+            node_generator.payment_start()
```

Comment:
let's add a `debug` log in here that we're using the `start` delay for the payment, could be useful.